lib/ecto/query/planner.ex

```diff
@@ -1913,7 +1913,7 @@ defmodule Ecto.Query.Planner do
       {:ok, v} ->
         {:ok, v}
       _ ->
-        {:error, "value `#{inspect v}` in `#{kind}` cannot be cast to type #{inspect type}"}
+        {:error, "value `#{inspect v}` in `#{kind}` cannot be cast to type #{Ecto.Type.format(type)}"}
```

Comment:
I went to everywhere in the code that some form of `inspect type` was called and changed almost all of them to this new signature. After this PR there will be some places where `inspect(type)` is still called because it felt like an overreach to push them down this path.

Reply:
Let me know @josevalim @wojtekmach if you would prefer that any instance of `inspect(type)` goes down this path

Reply:
@v0idpwn fixed two of those see 2295 explanation in my original comment

Reply:
Feeling good that I have all desired `inspect type` spots that are related to this
refactor covered.